core/domain/stats_services.py

```diff
@@ -16,251 +16,137 @@
-    for submitted_answer in submitted_answer_list:
-        submitted_answer.validate()
+        exp_id: str. ID of the exploration.
+        exp_version: int. Version of the exploration.
 
-    stats_models.StateAnswersModel.insert_submitted_answers(
-        state_answers.exploration_id, state_answers.exploration_version,
-        state_answers.state_name, state_answers.interaction_id,
-        state_answers.get_submitted_answer_dict_list())
+    Returns:
+        ExplorationStats. The domain object for exploration statistics.
+
+    Raises:
+        Exception: Entity for class ExplorationStatsModel with id not found.
+    """
+    model_id = exp_id + '.' + str(exp_version)
+    exploration_stats_model = stats_models.ExplorationStatsModel.get(
+        model_id)
+    exploration_stats = get_exploration_stats_from_model(
+        exploration_stats_model)
+    return exploration_stats
 
 
-def get_state_answers(exploration_id, exploration_version, state_name):
-    """Returns a StateAnswers object containing all answers associated with the
-    specified exploration state, or None if no such answers have yet been
-    submitted.
+def get_exploration_stats_from_model(exploration_stats_model):
+    """Gets an ExplorationStats domain object from an ExplorationStatsModel
+    instance.
 
     Args:
-        exploration_id: str. The exploration ID.
-        exploration_version: int. The version of the exploration to fetch
-            answers for.
-        state_name: str. The name of the state to fetch answers for.
+        exploration_stats_model: ExplorationStatsModel. Exploration statistics
+            model in datastore.
 
     Returns:
-        StateAnswers or None. A StateAnswers object containing all answers
-        associated with the state, or None if no such answers exist.
+        ExplorationStats. The domain object for exploration statistics.
     """
-    state_answers_models = stats_models.StateAnswersModel.get_all_models(
-        exploration_id, exploration_version, state_name)
-    if state_answers_models:
-        main_state_answers_model = state_answers_models[0]
-        submitted_answer_dict_list = itertools.chain.from_iterable([
-            state_answers_model.submitted_answer_list
-            for state_answers_model in state_answers_models])
-        return stats_domain.StateAnswers(
-            exploration_id, exploration_version, state_name,
-            main_state_answers_model.interaction_id,
-            [stats_domain.SubmittedAnswer.from_dict(submitted_answer_dict)
-             for submitted_answer_dict in submitted_answer_dict_list],
-            schema_version=main_state_answers_model.schema_version)
-    else:
-        return None
-
-
-def get_sample_answers(exploration_id, exploration_version, state_name):
-    """Fetches a list of sample answers that were submitted to the specified
-    exploration state (at the given version of the exploration).
+    new_state_stats_mapping = {}
```

Comment:
NB: Can do:

    new_state_stats_mapping = {
        state_name: stats_domain.StateStats.from_dict(...)
        for state_name in exploration_stats_model.state_stats_mapping
    }

if you like.

Reply:
done.